executor/table_reader.go

```diff
@@ -205,9 +205,25 @@ func (e *TableReaderExecutor) Close() error {
 	}
 	e.kvRanges = e.kvRanges[:0]
 	e.ctx.StoreQueryFeedback(e.feedback)
+	e.collectIndexUsage()
 	return err
 }
 
+func (e *TableReaderExecutor) collectIndexUsage() {
+	if e.ctx.IndexUsageCollectorActivated() {
```

Comment:
Should we skip this function if it is full scan?

Reply:
Yes, but we check if it is full scan blew.